eland/operations.py

```diff
@@ -626,14 +669,68 @@ def _unpack_metric_aggs(
                         else:
                             agg_value = field.np_dtype.type(agg_value)
 
-                values.append(agg_value)
+                if not percentile_values:
+                    values.append(agg_value)
 
             # If numeric_only is True and We only have a NaN type field then we check for empty.
             if values:
                 results[field.column] = values if len(values) > 1 else values[0]
+            # This only runs when df.quantile() or series.quantile() is called
+            if percentile_values and not is_dataframe_agg:
+                results[f"{field.column}"] = percentile_values
 
         return results
 
+    def quantile(
+        self,
+        query_compiler: "QueryCompiler",
+        pd_aggs: List[str],
+        quantiles: Union[int, float, List[int], List[float]],
+        is_dataframe: bool = True,
+        numeric_only: Optional[bool] = True,
+    ) -> Union[pd.DataFrame, pd.Series]:
+        # To verify if quantile range falls between 0 to 1
+        def quantile_to_percentile(quantile: Any) -> float:
+            if isinstance(quantile, (int, float)):
+                quantile = float(quantile)
+                if quantile > 1 or quantile < 0:
+                    raise ValueError(
+                        f"quantile should be in range of 0 and 1, given {quantile}"
+                    )
+            else:
+                raise TypeError("quantile should be of type int or float")
+            # quantile * 100 = percentile
+            # return float(...) because min(1.0) gives 1
+            return float(min(100, max(0, quantile * 100)))
+
+        percentiles = [
+            quantile_to_percentile(x)
+            for x in (
+                (quantiles,) if not isinstance(quantiles, (list, tuple)) else quantiles
+            )
+        ]
+
+        result = self._metric_aggs(
+            query_compiler,
+            pd_aggs=pd_aggs,
+            percentiles=percentiles,
+            is_dataframe_agg=False,
+            numeric_only=numeric_only,
+        )
+
+        df = pd.DataFrame(
+            result,
+            index=[i / 100 for i in percentiles],
```

Comment:
Should `index` be `quantiles` instead of another transformation?

Reply:
Umm, I am kind of confused here
because pandas is behaving like this 
If we give [0,1, 0.2] it is considering index as floats and dtype as float64
Otherwise if I give [0,1] it is becoming index as int and dtype as int64

So, either I leave the calculation to make it float

or write again `[quantiles] if isinstance(quantiles, float) else list(map(float,quantiles))`

So, I am just going to leave with calcuation....